src/states/node.rs

```diff
@@ -1654,6 +1736,64 @@ impl Node {
         true
     }
 
+    fn send_own_group_merge(&mut self,
+                            targets: Vec<XorName>,
+                            merge_details: OwnMergeDetails<XorName>,
+                            src: Authority) {
+        for target in &targets {
+            let (connected, needed): (Vec<_>, Vec<_>) = merge_details.groups
+                .clone()
+                .into_iter()
+                .partition(|&(prefix, _)| merge_details.prefix.is_compatible(&prefix));
```

Comment:
Yes, I did indeed mean `is_neighbour()`, not `is_compatible()`!

As for the second point, after discussion in Slack the outcome is that I plan to fix this in a separate PR as it involves non-trivial changes to the `RoutingTable`.

The approach will be the recipients of an `OwnGroupMerge` message call a new RT function `prepare_merge(OwnMergeDetails) -> OwnMergeDetails` and send this info to all of the merging groups.  On receipt, these nodes then call `RT::merge_own_group()` which is changed to have an optional return.  This will accumulate the incoming details, returning `None` each time until all expected groups have provided info, at which point it will return `Some` with the required `OtherGroupDetails` as now.